src/cudadev/CUDACore/device_unique_ptr.h

```diff
@@ -49,8 +49,8 @@ namespace cms {
 
     template <typename T>
     typename device::impl::make_device_unique_selector<T>::non_array make_device_unique(cudaStream_t stream) {
-      static_assert(std::is_trivially_constructible<T>::value,
-                    "Allocating with non-trivial constructor on the device memory is not supported");
+      static_assert(std::is_trivially_copyable<T>::value,
+                    "Allocating with non-trivial copy on the device memory is not supported");
```

Comment:
Maybe it is time to give up on "trivially constructible" requirement, it was somewhat annoying anyway.

I'm somewhat concerned on clarity that this function doesn't initialize the memory (with "trivially constructible" that's somewhat expected, although technically different from "value initialization" of `std::make_unique()`), but that's not a real issue.

With this change I would consider removing `make_device_unique_uninitialized()` (that really should have had the "trivially copyable" requirement).

Reply:
`make_device_unique_uninitialized()` is not used (neither in any variant of `pixeltrack standalone, nor in CMSSW (I checked `CMSSW_12_1_0_pre5`).

Both changes are fine in all variations of pixeltrack standalone: https://github.com/ericcano/pixeltrack-standalone/tree/device_unique_ptr_relax (should I PR this?)